src/main/java/seedu/address/logic/parser/transaction/EditTransactionCommandParser.java

```diff
@@ -65,20 +71,36 @@ public EditTransactionCommand parse(String args) throws ParseException {
             editTransactionDescriptor.setDateTime(
                     ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_DATETIME).get()));
         }
-        if (argMultimap.getValue(PREFIX_CATEGORY).isPresent()) {
-            editTransactionDescriptor.setCategory(
-                    ParserUtil.parseCategory(argMultimap.getValue(PREFIX_CATEGORY).get()));
-        }
         if (argMultimap.getValue(PREFIX_LOCATION).isPresent()) {
             editTransactionDescriptor.setLocation(
                     ParserUtil.parseLocation(argMultimap.getValue(PREFIX_LOCATION).get()));
         }
 
+        parseCategoriesForEdit(argMultimap.getAllValues(PREFIX_CATEGORY))
+                .ifPresent(editTransactionDescriptor::setCategories);
+
         if (!editTransactionDescriptor.isAnyFieldEdited()) {
             throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
         }
 
         return new EditTransactionCommand(index, editTransactionDescriptor);
     }
 
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Category>} if {@code categories} is non-empty.
+     * If {@code categories} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Category>} containing zero tags.
+     */
+    private Optional<Set<Category>> parseCategoriesForEdit(Collection<String> categories) throws ParseException {
+        assert categories != null;
```

Comment:
Any reason why this is an `assert` and not `requireNonNull`?

Reply:
I referenced AB3, but if I had to make a guess for their decision, it would probably be because the only place where they use this method would never lead to a null value being passed in here. But I guess we could make do with a `requireNonNull`.

Reply:
If it's not an issue, then we can keep the assertion